pkg/registry/csv.go

```diff
@@ -187,3 +191,74 @@ func (csv *ClusterServiceVersion) GetApiServiceDefinitions() (owned []*Definitio
 	required = definitions.Required
 	return
 }
+
+// GetRelatedImage returns the list of associated images for the operator
+func (csv *ClusterServiceVersion) GetRelatedImages() (imageSet map[string]struct{}, err error) {
+	var objmap map[string]*json.RawMessage
+	imageSet = make(map[string]struct{})
+
+	if err = json.Unmarshal(csv.Spec, &objmap); err != nil {
+		return
+	}
+
+	rawValue, ok := objmap[relatedImages]
+	if !ok || rawValue == nil {
+		return
+	}
+
+	type relatedImage struct {
+		Name string `json:"name"`
+		Ref string `json:"image"`
+	}
+	var relatedImages []relatedImage
+	if err = json.Unmarshal(*rawValue, &relatedImages); err != nil {
+		return
+	}
+
+	for _, img := range relatedImages {
+		imageSet[img.Ref] = struct{}{}
+	}
+
+	return
+}
+
+// GetOperatorImages returns a list of any images used to run the operator.
+// Currently this pulls any images in the pod specs of operator deployments.
+func (csv *ClusterServiceVersion) GetOperatorImages() (map[string]struct{}, error) {
+	type dep struct {
```

Comment:
Can we import these types from somewhere rather than redefine them here? If/When the spec changes this may be hard to find.

Reply:
This is specifically to avoid importing OLM